### ISSUE TITLE
password

### ISSUE THREAD

**Issue by aliafshany** [leonspok/Huawei4GSignalInfo#1]
hi 
this is an  awesome app but the thing is my modem has a password so if it is possible please be kind enough to add and option to enter password on "configure" section.

best

**Comment by leonspok** [comment]
Hi! 
Yes, I can try to add it. The problem is that I don't have this router anymore, so I won't be able to properly test it. But for me it was working even though I also had admin password set on the router. 
Potentially, it might require a very simple change, but before implementing it I need to make sure that my assumptions are right. Could you, please, post the result of this terminal command?

```
curl "<router_address>/api/device/signal"
# OR
curl -u "<username>:<password>" "<router_address>/api/device/signal"
```

**Comment by aliafshany** [comment]
i tried those commands but nothing happened, i guess it is the address which is incorrect to get the signals.
my router is a little different than yours. it's Huawei 4g E5785Lh-22c but still scripts work on that for example [this script](https://github.com/littlejo/huawei-lte-examples)   for me looks like this:

<img width="451" alt="Screen Shot 2021-08-11 at 22 43 35" src="https://user-images.githubusercontent.com/40741745/129081522-e78c6ebc-6a95-4922-9597-2bd7ac10fc9b.png">

this image is the result of the commands you told (  please take into consideration that my router address is 192.168.1.1 and still those commands don't work)
<img width="455" alt="Screen Shot 2021-08-11 at 22 45 50" src="https://user-images.githubusercontent.com/40741745/129081806-f8539bf4-3263-4466-9d56-b3edddaddc8f.png">



and this is the address for signals

![Screen Shot 2021-08-11 at 22 46 17](https://user-images.githubusercontent.com/40741745/129081934-162ffc81-1232-4445-94e2-2dfbbe02a183.png)


if you think adding password option and adjusting it to this kind of router is so much time and energy consumer please feel free to deny doing that.
best

**Comment by leonspok** [comment]
Thank you for the script! I'll check it.

Could you, please, try again `curl` but without "<" and ">". Something like this: 

```bash
curl -u "admin:2741239266" "http://192.168.1.1/api/device/signal"
```

**Comment by aliafshany** [comment]
oh i'm really sorry to put those <> in the address :)

sure i can do that.
these are the results:

<img width="662" alt="Screen Shot 2021-08-11 at 23 02 47" src="https://user-images.githubusercontent.com/40741745/129083931-db4fd3f6-b226-4ceb-942b-005f4604d03f.png">

**Comment by leonspok** [comment]
Yeah, it complains about missing CSRF token. Although it shouldn't because `/api/device/signal` doesn't require authorisation ([huawei-lte-api](https://github.com/Salamek/huawei-lte-api) also states that). What you will see if you open `http://admin:2741239266@192.168.1.1/api/device/signal` in your browser?

**Comment by aliafshany** [comment]
here you are sir:

![Screen Shot 2021-08-11 at 23 18 41](https://user-images.githubusercontent.com/40741745/129086156-3900e0b1-6eec-413f-8f61-27452bfdae13.png)
![Screen Shot 2021-08-11 at 23 19 11](https://user-images.githubusercontent.com/40741745/129086169-acc180fe-dd59-406d-8542-7f3507915c66.png)

**Comment by leonspok** [comment]
Ok, then it means that this will require a bit more work than I expected and it will be hard to do without physical device in my hands. 

If you want to try to implement it yourself, then I would suggest to start from `SignalInfoFetcher.swift`. You will need to add support for CSRF tokens there.

**Comment by aliafshany** [comment]
oh that is unfortunate. btw i love the way your app looks like, this is why i opened an issue.
thank you for replying me back so fast and i appreciate your time

best
